test/new-e2e/tests/agent-shared-components/api/api_test.go

```diff
@@ -406,15 +404,22 @@ func (v *apiSuite) TestDefaultAgentAPIEndpoints() {
 	authtokenContent := v.Env().RemoteHost.MustExecute("sudo cat " + authTokenFilePath)
 	authtoken := strings.TrimSpace(authtokenContent)
 
+	host := v.Env().RemoteHost
 	for _, testcase := range testcases {
-		cmd := testcase.fetchCommand(authtoken)
-		host := v.Env().RemoteHost
 		v.T().Run(fmt.Sprintf("API - %s test", testcase.name), func(t *testing.T) {
+			req, err := testcase.httpRequest(authtoken)
+			assert.NoError(t, err, "failed to create request")
+
+			hostHTTPClient := host.NewHTTPClient()
 			require.EventuallyWithT(t, func(ct *assert.CollectT) {
-				resp, err := host.Execute(cmd)
-				if assert.NoError(ct, err) {
-					testcase.assert(ct, testcase.agentEndpointInfo, resp)
-				}
+				resp, err := hostHTTPClient.Do(req)
+				assert.NoError(ct, err, "failed to send request")
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Should we check the http status code ? (it depends on the tests I guess)
`Do` only returns an error if there's no response / the response is invalid, but it doesn't complain when the response is a 4xx or 5xx

Reply:
Same comment for other instances below

Reply:
fixed in https://github.com/DataDog/datadog-agent/pull/27685/commits/0e92ac6f1f91831ff126803482cb50df8e64962e